src/staking/manager.ts

```diff
@@ -0,0 +1,660 @@
+      btcSig: ecdsaSig,
+    };
+
+    // Prepare and send protobuf message
+    const msg: btcstakingtx.MsgCreateBTCDelegation =
+      btcstakingtx.MsgCreateBTCDelegation.fromPartial({
+        stakerAddr: bech32Address,
+        pop: proofOfPossession,
+        btcPk: Uint8Array.from(
+          Buffer.from(stakerInfo.publicKeyNoCoordHex, "hex"),
+        ),
+        fpBtcPkList: [
+          Uint8Array.from(
+            Buffer.from(stakingInput.finalityProviderPkNoCoordHex, "hex"),
+          ),
+        ],
+        stakingTime: stakingInput.stakingTimelock,
+        stakingValue: stakingInput.stakingAmountSat,
+        stakingTx: Uint8Array.from(stakingTx.toBuffer()),
+        slashingTx: Uint8Array.from(
+          Buffer.from(clearTxSignatures(signedSlashingTx).toHex(), "hex"),
+        ),
+        delegatorSlashingSig: Uint8Array.from(slashingSig),
+        unbondingTime: param.unbondingTime,
+        unbondingTx: Uint8Array.from(unbondingTx.toBuffer()),
+        unbondingValue: stakingInput.stakingAmountSat - param.unbondingFeeSat,
+        unbondingSlashingTx: Uint8Array.from(
+          Buffer.from(
+            clearTxSignatures(signedUnbondingSlashingTx).toHex(),
+            "hex",
+          ),
+        ),
+        delegatorUnbondingSlashingSig: Uint8Array.from(unbondingSignatures),
+        stakingTxInclusionProof: inclusionProof,
+      });
+
+    return {
+      typeUrl: BBN_REGISTRY_TYPE_URLS.MsgCreateBTCDelegation,
+      value: msg,
+    };
+  };
+
+  private getInclusionProof(
+    inclusionProof: InclusionProof,
+  ): btcstaking.InclusionProof {
+    const {
+      pos,
+      merkle,
+      blockHash
+    } = inclusionProof;
+    const proofHex = deriveMerkleProof(merkle);
+  
+    const hash = reverseBuffer(Uint8Array.from(Buffer.from(blockHash, "hex")));
+    const inclusionProofKey: btccheckpoint.TransactionKey =
+      btccheckpoint.TransactionKey.fromPartial({
+        index: pos,
+        hash,
+      });
+    return btcstaking.InclusionProof.fromPartial({
+      key: inclusionProofKey,
+      proof: Uint8Array.from(Buffer.from(proofHex, "hex")),
+    });
+  };
+}
+
+/**
+ * Extracts the first valid Schnorr signature from a signed transaction.
+ * @param singedTransaction - The signed transaction.
+ * @returns The first valid Schnorr signature or undefined if no valid signature is found.
+ */
+export const extractSchnorrSignaturesFromTransaction = (
+  singedTransaction: Transaction,
+): Buffer | undefined => {
+  // Loop through each input to extract the witness signature
+  for (const input of singedTransaction.ins) {
+    if (input.witness && input.witness.length > 0) {
+      const schnorrSignature = input.witness[0];
+
+      // Check that it's a 64-byte Schnorr signature
+      if (schnorrSignature.length === 64) {
+        return schnorrSignature; // Return the first valid signature found
+      }
+    }
+  }
+  return undefined;
+};
+
+/**
+ * Clears the signatures from a transaction.
```

Comment:
What do you mean by `clear`? Why are we doing this? Docstring should detail this.